main.go

```diff
@@ -40,118 +42,209 @@ import (
 		if err != nil {
 			flag.Usage()
 			os.Exit(1)
 		}
 		processConfig(ossIndexConfig)
+		Logger.Info("Nancy finished parsing config for OSS Index")
 	}
 }
 
 func printHeader(print bool) {
 	if print {
+		Logger.Info("Attempting to print header")
 		figure.NewFigure("Nancy", "larry3d", true).Print()
 		figure.NewFigure("By Sonatype & Friends", "pepper", true).Print()
+
+		Logger.WithField("version", buildversion.BuildVersion).Info("Printing Nancy version")
 		log.Println("Nancy version: " + buildversion.BuildVersion)
+		Logger.Info("Finished printing header")
 	}
 }
 
 func processConfig(config configuration.Configuration) {
 	if config.Help {
+		Logger.Info("Printing usage and exiting clean")
 		flag.Usage()
 		os.Exit(0)
 	}
 
 	if config.Version {
+		Logger.WithFields(logrus.Fields{
+			"build_time":   buildversion.BuildTime,
+			"build_commit": buildversion.BuildCommit,
+			"version":      buildversion.BuildVersion,
+		}).Info("Printing version information and exiting clean")
+
 		fmt.Println(buildversion.BuildVersion)
 		_, _ = fmt.Printf("build time: %s\n", buildversion.BuildTime)
 		_, _ = fmt.Printf("build commit: %s\n", buildversion.BuildCommit)
 		os.Exit(0)
 	}
 
+	if config.Info || config.Debug || config.Trace {
+		if config.Info {
+			Logger.Level = logrus.InfoLevel
+		}
+		if config.Debug {
+			Logger.Level = logrus.DebugLevel
+		}
+		if config.Trace {
+			Logger.Level = logrus.TraceLevel
+		}
+	}
+
 	if config.CleanCache {
+		Logger.Info("Attempting to clean cache")
 		if err := ossindex.RemoveCacheDirectory(); err != nil {
+			Logger.WithField("error", err).Error("Error cleaning cache")
 			fmt.Printf("ERROR: cleaning cache: %v\n", err)
 			os.Exit(1)
 		}
+		Logger.Info("Cache cleaned")
 		return
 	}
 
 	if config.Quiet {
+		Logger.Debug("Setting console log output to discard, quiet requested")
 		log.SetOutput(ioutil.Discard)
 	}
 
 	printHeader((!config.Quiet && reflect.TypeOf(config.Formatter).String() == "*audit.AuditLogTextFormatter"))
 
 	if config.UseStdIn {
+		Logger.Info("Parsing config for StdIn")
 		doStdInAndParse(config)
 	}
 	if !config.UseStdIn {
+		Logger.Info("Parsing config for file based scan")
 		doCheckExistenceAndParse(config)
 	}
 }
 
 func processIQConfig(config configuration.IqConfiguration) {
+	// TODO: a lot of this code is a duplication of the OSS Index config, probably should extract some of it
 	if config.Help {
+		Logger.Info("Printing usage and exiting clean")
 		flag.Usage()
 		os.Exit(0)
 	}
 
 	if config.Version {
+		Logger.WithFields(logrus.Fields{
+			"build_time":   buildversion.BuildTime,
+			"build_commit": buildversion.BuildCommit,
+			"version":      buildversion.BuildVersion,
+		}).Info("Printing version information and exiting clean")
+
 		fmt.Println(buildversion.BuildVersion)
 		_, _ = fmt.Printf("build time: %s\n", buildversion.BuildTime)
 		_, _ = fmt.Printf("build commit: %s\n", buildversion.BuildCommit)
 		os.Exit(0)
 	}
 
+	if config.Info || config.Debug || config.Trace {
```

Comment:
minor: Why this outer if?

Maybe refactor into shared method for both OSSI and IQ? (or not...)

Reply:
Right now I can't do that too easy, config has two different structs, and no common interface. Gonna have to leave it like this until we simplify config a bit (something I'd like to do)

Reply:
Doh! That makes total sense.